[전은형] Mission 4 : 로그인 기능 구현

ooooo5h opened the pull request:
## 수정 사항 간략한 한줄 요약
- 로그인 기능 구현
		 
## 수정 사항들 자세한 내용
- 클라이언트로부터 이메일, 비밀번호 입력받기
- 이메일이나 비밀번호가 없다면 KEY_ERROR, 400 반환
- 잘못된 이메일이면 INVALID_USER, 401 반환
- 잘못된 비밀번호면 INVALID_USER, 401 반환
- 로그인 성공 시 SUCCESS, 200 반환
- 로그인 url 생성

## 기타 질문 및 특이 사항

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
조건을 만족한다면 각 빈칸에 x 표시 해주세요
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

ooooo5h committed:
login not finished

ooooo5h committed:
create loginview

ooooo5h committed:
modified signin and urls

ooooo5h committed:
solved git merge conflict

fkelfk commented on students/eunhyung/users/views.py:
```diff
@@ -40,10 +40,20 @@ def post(self, request):
+                return JsonResponse({'message':'INVALID_USER'}, status=401)
+            
+            if data['password'] != email_exist_user.password:
+                return JsonResponse({'message':'INVALID_USER'}, status=401)
```
혹시 데이터베이스 상에 있는 무작위 이메일과 무작위 비밀번호를 입력하면 어떤 결과가 나오나요?

ooooo5h replied:
User 객체들 중에서 DB에 들어가있는 email의 값이 클라이언트로부터 입력받은 email의 값과 같은 객체를 email_exist_user 변수에 저장했습니다. 그 다음 DB에 email이 존재하는 사용자의 DB상 저장된 비밀번호와 클라이언트로부터 입력받은 password의 값이 같지 않다면 INVALID_USER 메시지를 반환합니다. 그래서 이메일과 비밀번호가 일치하지 않는다면 401에러를 반환합니다.

fkelfk reviewed (commented):
혹시 무작위 이메일과 무작위 비밀번호를 입력하면 어떤 결과가 나오나요?

WorkYong commented on students/eunhyung/users/views.py:
```diff
@@ -40,10 +40,20 @@ def post(self, request):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            email_exist_user = User.objects.get(email=data['email'])
```
Get을 이용하게 되면 email을 틀리게되면 리턴값이 Invaild_users가 안되지 않나요? 결과값이 궁금합니다.

ooooo5h replied:
그러네요 try, except 사용해서 처리했는데도 의도한 대로 에러처리가 되지 않네요 부들부들 수정해보도록 하겠습니다

ooooo5h replied:
@WorkYong 정용님 덕분에 코드의 잘못된 점을 수정할 수 있었습니다 감사해요!! 🙏

ooooo5h committed:
SignInView 수정중
 - 그냥 except로 처리했더니 이메일/비밀번호가 없는 경우 KEY_ERROR로 발생
 - KeyError를 추가함: 입력값 없으면 KeyError 잘 발생
 - DB에 없는 유저 정보를 입력한 경우 User.DoesNotExist가 발생하게 에러 추가
 - 하지만 비밀번호/이메일 없는 경우 KEY_ERROR가 발생??

ooooo5h committed:
잘못된 이메일만 입력시 KEY_ERROR로 빠지지않고 INVALID_USER로 빠지는 이유

ooooo5h committed:
SignInView 수정 완료

ooooo5h committed:
줄맞춤

ooooo5h committed:
회원가입뷰 except -> except KeyError로 수정

Kang-kyunghun commented on students/eunhyung/users/views.py:
```diff
@@ -37,13 +37,26 @@ def post(self, request):
+            
+            email_exist_user = User.objects.get(email=email)
+                        
+            if password == email_exist_user.password:
+                return JsonResponse({'message':'SUCCESS'}, status=200)
+            else:
+                return JsonResponse({'message':'INVALID_USER'}, status=401)
```
일치하지 않는 경우부터 생각한다면 else을 사용하지 않고 조금 더 가독성을 높일 수 있을 것 같습니다.
```suggestion
            if password != email_exist_user.password:
                return JsonResponse({'message':'INVALID_USER'}, status=401)

            return JsonResponse({'message':'SUCCESS'}, status=200)
```

ooooo5h replied:
앗 넵!! try문 내부에서 로그인에 실패하는 경우를 먼저 if문으로 걸러내고 나서 로그인을 처리하도록 수정 사항 반영했습니다!

Kang-kyunghun commented:
리뷰도 잘 달아주시고, 리뷰에 대해 답변도 잘 해주셨습니다!!

ooooo5h committed:
SignInView 내부 조건문 순서 수정하여 가독성 높임

ooooo5h committed:
SignInView 내부 주석 삭제
